### ISSUE TITLE
Recreate DeepSense.io's ConvNet

### ISSUE THREAD

**Issue by marcinwrochna** [VeryDisappointedSalad/Prox-LoRA#10]
Recreate the architecture described in https://deepsense.ai/blog/diagnosing-diabetic-retinopathy-with-deep-learning/
and try to achieve somewhat similar results.

Or this (where they specify learning rates etc.): https://www.kaggle.com/competitions/diabetic-retinopathy-detection/writeups/o-o-team-o-o-solution-summary

This is only about the ConvNet.
Later, we can optionally try the other stuff they and others did (in order from most likely to give cheap benefits):
* [histogram equalization](https://docs.pytorch.org/vision/main/generated/torchvision.transforms.functional.equalize.html) input transform.
* LeakyRelu with alpha=1/3 or 1/2.
* use 512×512 resolution.
* RandomCrop (448×448)
* loss weights to counter class imbalance (0=73.48%, 1=6.96%, 2=15.07%, 3=2.48%, 4=2.01%), or undersampling class zero.
* [continuous kappa loss](https://defauw.ai/diabetic-retinopathy-detection/#the-kappa-metric)
* various color augmentations.
* test-time augmentation
* eye blending (blending predictions from left and right eye).
* optimizing a custom discretizer for the logits (like thresholded weighted sum)
* kappa optimization.
